switchio/handlers.py

```diff
@@ -321,9 +292,10 @@ def _handle_answer(self, e):
             self.log.warn('Skipping answer of {}'.format(uuid))
             return False, None
 
-    @handler('CHANNEL_HANGUP')
-    def _handle_hangup(self, e):
-        '''Handle hangup events
+    @handler('CHANNEL_DESTROY')
```

Comment:
Not sure of the context here, but be aware destroy may come much later than hangup (several seconds later, depending on the registered handlers for hangup, such as CDR)

Reply:
Yes that's fine the whole point of this handler is really as a `Session` model destructor.

If coroutine code wants to await `CHANNEL_HANGUP` it can but the `Session` model should be destroyed when the FS minion says it is. This avoids situations where a `CHANNEL_HANGUP` comes before another app related event (like `PLAYBACK_STOP`) and we want to await them in the expected order.